### ISSUE TITLE
UBERMAG installation problem in desktop (apple M2 chip)

### ISSUE THREAD

**Issue by DebanjanPolley** [ubermag/help#260]
I was previously using UBERMAG on a Mac (with an Intel chip) and everything was working fine. Now, I am trying to install UBERMAG on a desktop (Mac M2 chip) and facing problems, can you please help?

I have installed Anaconda for (Mac M2 chip). (https://www.anaconda.com/download#macos)
Anaconda got installed properly. 
Then I activated a new environment using `conda create -n UM1 python=3.10`, it went well.
Now, when I am trying to install UBERMAG using `conda install --channel conda-forge ubermag` in the UM1 envs, the following errors come up.

`Collecting package metadata (current_repodata.json): done
Solving environment: unsuccessful initial attempt using frozen solve. Retrying with flexible solve.
Solving environment: unsuccessful attempt using repodata from current_repodata.json, retrying with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: unsuccessful initial attempt using frozen solve. Retrying with flexible solve.
Solving environment: - 
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed                                                                          

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

(UM1) debanjanpolley@Mac-mini ~ % 
`

What am I doing wrong?

**Comment by lang-m** [comment]
Hi @DebanjanPolley,
the error is presumably coming from OOMMF for which we currently do not provide a conda package for apple M1/M2 (see also #140).
As a workaround, you can manually install OOMMF (e.g. from here: github.com/fangohr/oommf) and then install Ubermag using pip. Additional details are provided here: https://ubermag.github.io/installation.html#advanced-installation

**Comment by jedcheng** [comment]
I don't think it is possible to compile oommf for Arm. There is not a platform for Arm Mac in tcl

**Comment by DebanjanPolley** [comment]
Recently, I found an easier way to install UBERMAG on a Mac (M1/M2 chip). 

I had to build an osx-64 environment within my M2 Mac in the following way:

1. Create an osx-64 virtual environment `ubermag_env` 
`CONDA_SUBDIR=osx-64 conda create -n ubermag_env python`
2. activate the virtual environment
`conda activate ubermag_env` 
3. config osx-64
`conda config --env --set subdir osx-64` 

Then use `$ conda install --channel conda-forge ubermag` to install UBERMAG
